src/Cms/Api.php

```diff
@@ -81,19 +79,7 @@ public function fieldApi($model, string $name, string $path = null)
      */
     public function file(string $path = null, string $filename)
     {
-        $filename = urldecode($filename);
-        $file     = $this->parent($path)->file($filename);
-
-        if ($file && $file->isReadable() === true) {
-            return $file;
-        }
-
-        throw new NotFoundException([
-            'key'  => 'file.notFound',
-            'data' => [
-                'filename' => $filename
-            ]
-        ]);
+        return Panel::file($path, $filename);
```

Comment:
To be honest I feel like this is a bit of a weird mix. The API suddenly depends on the Panel, even though it has been the other way around before and partly still is. The added methods in the `Panel` class seem generally useful, so maybe they could get a new home in the CMS namespace.

Unrelated, but still kind of related: I feel like the `Panel\Panel` class is a bit chaotic from an architectural standpoint. It only has static methods even though basically all methods rely on the `$kirby` object, some even implicitly (using the global `kirby()` helper). I think a singleton class that gets instantiated with the `$kirby` object would be a lot cleaner and more future-proof.

Reply:
Yeah, I was thinking about that as well. Maybe we could introduce a new Kirby\Cms\Finder class with methods like:`Finder::file()`

The architecture of the Panel class really grew a lot. There's definitely room for improvements. The singleton could be a good idea. But I wouldn't solve this in this PR.

Reply:
I've created an issue: https://github.com/getkirby/kirby/issues/3404